tests/test_removedir.py

```diff
@@ -0,0 +1,160 @@
+        api._authorities['auth1'].fs.open('archive2' 'r')
+
+    assert api.listdir('', authority_name='auth1') == []
+
+
+@pytest.mark.remove_dir
+def test_remove_dir_multi_versions_remove(api, auth1):
+
+    cache = TempFS()
+    api.attach_authority('auth1', auth1)
+    api.attach_cache(cache)
+
+    with open('test_file.txt', 'w+') as f:
+        f.write(u'this is an upload test')
+
+    var = api.create('archive1', authority_name='auth1', versioned=True)
+    var.update('test_file.txt', cache=True)
+
+    with var.open('w') as f:
+        f.write(u'update update')
+
+    assert len(var.get_versions()) == 2
+
+    var.delete()
+
+    with pytest.raises(KeyError):
+        api.get_archive('archive1')
+
+    with pytest.raises(ResourceNotFoundError):
+        api._authorities['auth1'].fs.open('archive1', 'r')
+
+    with pytest.raises(ResourceNotFoundError):
+        api.cache.fs.open('archive1', 'r')
+
+    assert api.listdir('', authority_name='auth1') == []
+
+
+@pytest.mark.remove_dir
+def test_multi_api(api1, api2, auth1, cache1, cache2, opener):
+    '''
+    Test upload/download/cache operations with two users
+    '''
+    # Create two separate users. Each user connects to the
+    # same authority and the same manager table (apis are
+    # initialized with the same manager table but different
+    # manager instance objects). Each user has its own
+    # separate cache.
+
+    api1.attach_authority('auth1', auth1)
+    api1.attach_cache(cache1)
+
+    api2.attach_authority('auth1', auth1)
+    api2.attach_cache(cache2)
+
+    with open('text_file.txt', 'w') as f:
+        f.write(u'Stay Stoked')
+
+    archive1 = api1.create('myArchive', versioned=False)
+    archive1.update('text_file.txt')
+
+    # Turn on caching in archive 1 and assert creation
+
+    archive1.cache()
+    assert archive1.is_cached() is True
+    assert archive1.api.cache.fs is cache1
+
+    with opener(archive1, 'w+') as f1:
+        f1.write(u('Be happy and Stay Stoked'))
+
+    assert auth1.isfile('myArchive')
+    assert cache1.isfile('myArchive')
+
+    archive2 = api2.get_archive('myArchive')
+
+    # Turn on caching in archive 2 and assert creation
+    archive2.cache()
+    assert archive2.is_cached() is True
+    assert archive2.api.cache.fs is cache2
+
+    with archive2.open('r') as f:
+        assert u(f.read()) == u'Be happy and Stay Stoked'
+
+    archive2.delete()
+
+    with pytest.raises(ResourceNotFoundError):
+        api2._authorities['auth1'].fs.open('myArchive', 'r')
+
+    with pytest.raises(ResourceNotFoundError):
+        api2.cache.fs.open('myArchive', 'r')
+
+    assert cache1.isfile('myArchive')
+
+    with pytest.raises(ResourceNotFoundError):
+        api1._authorities['auth1'].fs.open('myArchive', 'r')
+
+    # using remove_from_cache because this archive is no longer in the manager
+    archive1.remove_from_cache()
+
+    with pytest.raises(ResourceNotFoundError):
+        api1.cache.fs.open('myArchive', 'r')
```

Comment:
If we ever get around to restructuring our tests, it may be a good idea to test against a wider variety of archive names. I do a lot of testing with 'archive1.txt'-like archive names, and this doesn't test the automatic folder creation (and non-deletion) behavior. I'd like to test things like 'my/long/archive/name'.

But these tests are fantastic and given the overlap with the doctests in the examples I'm satisfied that this is working.